components/library/src/content/mod.rs

```diff
@@ -46,18 +47,11 @@ pub fn find_related_assets(path: &Path, config: &Config) -> Vec<PathBuf> {
     }
 
     if let Some(ref globset) = config.ignored_content_globset {
-        // `find_related_assets` only scans the immediate directory (it is not recursive) so our
-        // filtering only needs to work against the file_name component, not the full suffix. If
-        // `find_related_assets` was changed to also return files in subdirectories, we could
-        // use `PathBuf.strip_prefix` to remove the parent directory and then glob-filter
-        // against the remaining path. Note that the current behaviour effectively means that
-        // the `ignored_content` setting in the config file is limited to single-file glob
-        // patterns (no "**" patterns).
```

Comment:
The changes are centered around this advice. Above, I replaced `read_dir` by `WalkDir` to collect assets recursively. And everywhere where the collected assets are accessed I replaced `.file_name()` with `.strip_prefix(&page.file.path.parent().unwrap())`. Note that I wanted to use `.strip_prefix(&page.file.parent)` originally but there is a special case for pages, where `parent` is actually `grand_parent`, that breaks this.

Reply:
> there is a special case for pages, where parent is actually grand_parent, that breaks this.

Do you mean when you have an asset folder? That's expected then since otherwise the parent would refer to the page itself

Reply:
Yeah, that's what I meant.